CONTENTdm.js

```diff
@@ -0,0 +1,577 @@
+{
+	"translatorID": "f715a5a1-c362-47cf-b736-2cb2c882b852",
+	"label": "CONTENTdm",
+	"creator": "Emma Reisz",
+	"target": "/cdm/|/cdm4/",
+	"minVersion": "3.0.9",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsbv",
+	"lastUpdated": "2015-11-02 10:55:08"
+}
+
+/**
+	***** BEGIN LICENSE BLOCK *****
+
+	CONTENTdm translator; Copyright © 2015 Emma Reisz
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+/*
+  ContentDM is an OCLC product used by several hundred libraries, archives and museums.
+  CDM may be hosted by OCLC or by the customer.
+  CDM includes an API which calls metadata as XML or JSON, but customers may disable it.
+  By preference, this translator scrapes XML from an API query (slow)
+  and from a webpage as a fallback (quick but vulnerable to a redesign).
+  XML is preferred to JSON because it allows code to be shared with the webscraper.
+  The translator will also scrape some data from legacy CDM webpages (version < 6).
+  Older CDM versions had no standardised field names so comprehensive scraping is unrealistic.
+*/
+
+function detectWeb( doc, url ) {
+  if ( !url.match( "/cdm/" ) && !url.match( "/cdm4/" ) ) return;
+  if ( url.match( "/cdm/search/" ) ) return "multiple";
+  else if ( url.match( "/cdm/landingpage" ) || !url.substring( url.indexOf( '/cdm/' ) + 5 ) ) return "webpage";
```

Comment:
Not sure what this is supposed to be. We generally don't match on plain web pages. Also, we prefer to use `url.indexOf("foo") != -1` or, if you need regexp, `/foo?/.test(url)` over `.match()`. Lastly, it's preferable to be more obvious with what you're trying to do. E.g. `!url.substring( url.indexOf( '/cdm/' ) + 5 )` should be `/\/cdm\/$/.test(url)` or `url.indexOf('/cdm/') == url.length - 5`